File: gpt-3.5-finetuned/stage2/declare_directive_device_resident_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
int test1() {
    int err = 0;
    srand(SEED);
    real_t* a = new real_t[n];
    real_t* b = new real_t[n];
    real_t* c = new real_t[n];

    for (int x = 0; x < n; ++x) {
        a[x] = rand() / (real_t)(RAND_MAX / 10);
        b[x] = rand() / (real_t)(RAND_MAX / 10);
        c[x] = 0.0;
    }

    #pragma acc declare create(a[0:n]) device_resident 
    #pragma acc data copyin(a[0:n], b[0:n]) copyout(c[0:n])
    {
        #pragma acc parallel
        {
            #pragma acc loop
            for (int i = 0; i < n; ++i) {
                a[i] += b[i];
            }
        }
    }

    for (int x = 0; x < n; ++x) {
        if (fabs(a[x] - (b[x])) > PRECISION) {
            err += 1;
        }
    }

    return err;
}
#endif

int main() {
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x) {
        failed = failed + test1();
    }
    if (failed != 0) {
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-finetuned/stage2/declare_directive_device_resident_clause.f90
!T1:declare directive device_resident clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0

        REAL(8), DIMENSION(LOOPCOUNT), DEVICE_RESIDENT :: a, b
        REAL(8), DIMENSION(LOOPCOUNT) :: c
        INTEGER i

        SEEDDIM(1) = 1
        SEEDDIM(1) = SEED
        CALL RANDOM_SEED(PUT=SEEDDIM)

        ! complete test here

        CALL RANDOM_NUMBER(a)
        CALL RANDOM_NUMBER(b)
        CALL RANDOM_NUMBER(c)

        !$acc declare create(a(1:LOOPCOUNT), b(1:LOOPCOUNT))

        !$acc parallel loop reduction(+:errors)
        DO i = 1, LOOPCOUNT
          a(i) = a(i) + c(i)
          b(i) = a(i) + c(i)
          c(i) = a(i) + b(i)
        END DO

        !$acc end parallel loop

        DO i = 1, LOOPCOUNT
          IF (abs(a(i) - (c(i) / 2)) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
          IF (abs(b(i) - (c(i) / 2)) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
          IF (abs(c(i) - (a(i) + b(i))) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM